t/055-subreq-vars.t

```diff
@@ -65,8 +65,8 @@ M(http-subrequest-start) {
 --- request
 GET /lua
 --- response_body_like: 500 Internal Server Error
---- error_log chop
-variable "dog" cannot be assigned a value (maybe you forgot to define it first?)
+--- error_log eval
+qr/variable "(dog|cat)" cannot be assigned a value \(maybe you forgot to define it first\?\)/
```

Comment:
I've modified test:
```
# vim:set ft= ts=4 sw=4 et fdm=marker:
use lib 'lib';
use Test::Nginx::Socket::Lua;

plan tests => repeat_each() * (blocks() * 3);
run_tests();

__DATA__

=== TEST 1: set non-existent variables via "vars" option
--- config
    location /other {
        echo Ok;
    }

    location /lua {
        content_by_lua '
            res = ngx.location.capture("/other",
                { vars = { dog = "hello", cat = 32 }});

            ngx.print(res.body)
        ';
    }

--- request
GET /lua
--- no_error_log
[error]
[alert]
```
And this shows:
```
% prove -r -l --directives t/055-subreq-vars-1.t
t/055-subreq-vars-1.t .. 1/3 
#   Failed test 'TEST 1: set non-existent variables via "vars" option - status code ok'
#   at lib/Test/Nginx/Socket.pm line 894.
#          got: '500'
#     expected: '200'

#   Failed test 'TEST 1: set non-existent variables via "vars" option - pattern "[error]" should not match any line in error.log but matches line "2015-03-04T14:04:30.674+00:00 [error] 16147\#0: *1 variable \"cat\" cannot be assigned a value (maybe you forgot to define it first?) , client: 127.0.0.1, server: localhost, request: \"GET /lua HTTP/1.1\", host: \"localhost\"" (req 0)'
#   at lib/Test/Nginx/Socket.pm line 1160.

#   Failed test 'TEST 1: set non-existent variables via "vars" option - pattern "[error]" should not match any line in error.log but matches line "2015-03-04T14:04:30.674+00:00 [error] 16147\#0: *1 lua entry thread aborted: runtime error: content_by_lua(nginx.conf:53):2: failed to adjust the subrequest: -1" (req 0)'
#   at lib/Test/Nginx/Socket.pm line 1160.
t/055-subreq-vars-1.t .. Failed 3/3 subtests 

Test Summary Report
-------------------
t/055-subreq-vars-1.t (Wstat: 0 Tests: 4 Failed: 4)
  Failed tests:  1-4
  Parse errors: Bad plan.  You planned 3 tests but ran 4.
Files=1, Tests=4,  0 wallclock secs ( 0.04 usr  0.01 sys +  0.20 cusr  0.07 csys =  0.32 CPU)
Result: FAIL
```
So my patch is correct. Error message is generated when vars table is assigned and the table is unordered.